### PR TITLE
Implement clock related functions in wasi

### PR DESCRIPTION
**Pull request by zherczeg**
Improve pointer handling of wasi functions.

### CODE REVIEW

File: src/wasi/WASI.cpp [hunk 1]
```diff
@@ -120,77 +175,73 @@ void WASI::fd_seek(ExecutionState& state, Value* argv, Value* result, Instance*
     uint32_t fd = argv[0].asI32();
     uint64_t fileDelta = argv[1].asI32();
     uint32_t whence = argv[2].asI32();
-    uint64_t newOffset = argv[3].asI32();
-
-    uvwasi_filesize_t out_addr = *(instance->memory(0)->buffer() + newOffset);
+    uvwasi_filesize_t* file_size = reinterpret_cast<uvwasi_filesize_t*>(get_memory_pointer(instance, argv[3], sizeof(uvwasi_filesize_t)));
```
Comment (clover2123) [inline src/wasi/WASI.cpp:178]: What if `get_memory_pointer` returns a nullptr here?
In WASI functions, when accessing memory, is range always correct?

File: src/wasi/WASI.cpp [hunk 2]
```diff
@@ -120,77 +175,73 @@ void WASI::fd_seek(ExecutionState& state, Value* argv, Value* result, Instance*
...
+    char* uvEnvironBuf = reinterpret_cast<char*>(get_memory_pointer(instance, argv[1], size));
+
+    if (uvEnviron == nullptr || uvEnvironBuf == nullptr) {
+        result[0] = Value(WasiErrNo::inval);
+        return;
+    }
 
-    char** uvEnviron = reinterpret_cast<char**>(instance->memory(0)->buffer() + env);
-    char* uvEnvironBuf = reinterpret_cast<char*>(instance->memory(0)->buffer() + environBuf);
+    TemporaryData pointers(count * sizeof(void*));
```
Comment (clover2123) [inline src/wasi/WASI.cpp:221]: It seems that `TemporaryData` structure is used only here,
so what about simply allocating a temporal buffer inside this `environ_get` function?

Comment (zherczeg) [reply src/wasi/WASI.cpp:221]: #308 should also use this. Sharing code should reduce the possible errors.

**Comment by zherczeg** [comment]
It looks like macos fails with "Error: The `brew link` step did not complete successfully".

File: src/wasi/WASI.cpp (hunk 2 above)
Comment (clover2123) [reply src/wasi/WASI.cpp:221]: I see

**Comment by clover2123** [comment]
> It looks like macos fails with "Error: The brew link step did not complete successfully".

This might be triggered by `pkg-config` installation in `macos-latest` env.
How about skipping `pkg-config` install in macos-latest(build-on-macos-arm64) ?

**Comment by zherczeg** [comment]
Thank you for the suggestion! The patch is working now

**Review by clover2123 — APPROVED** [review]
LGTM

**Pull request merged**